t/sanity.t

```diff
@@ -1966,3 +1968,479 @@ init_worker_by_lua_block {
+    listen 12359;
+    location = /status {
+        return 200;
+    }
+}
+
+server {
+    listen 12355;
+    listen 12357;
+    location = /status {
+        return 404;
+    }
+}
+
+server {
+    listen 12356;
+    location = /status {
+        return 503;
+    }
+}
+
+lua_shared_dict healthcheck 2m;
+access_by_lua_block {
+    ngx.shared.healthcheck:flush_all()
+    local hc = require "resty.upstream.healthcheck"
+    local upstream = require "ngx.upstream"
+
+    for _, upstream_name in ipairs{'foo.com', 'bar.com'} do
+        local ok, err = hc.spawn_checker{
+            shm = "healthcheck",
+            upstream = upstream_name,
+            type = "http",
+            http_req = "GET /status HTTP/1.0\r\nHost: localhost\r\n\r\n",
+            interval = 1000,  -- ms
+            concurrency = 2,
+            fall = 1,
+            dedup = true
+        }
+
+        -- This is necessary so the upstreams are health checked one after the other
+        ngx.sleep(0.01)
+
+        if not ok then
+            ngx.log(ngx.ERR, "failed to spawn health checker: ", err)
+            return
+        end
+    end
+}
+
+}
+--- config
+    location = /t {
+        access_log off;
+        content_by_lua '
+            ngx.sleep(0.52)
+
+            local hc = require "resty.upstream.healthcheck"
+            ngx.print(hc.status_page())
+
+            for i = 1, 4 do
+                local res = ngx.location.capture("/proxy")
+                ngx.say("upstream addr: ", res.header["X-Foo"])
+            end
+        ';
+    }
+
+    location = /proxy {
+        proxy_pass http://foo.com/;
+        header_filter_by_lua '
+            ngx.header["X-Foo"] = ngx.var.upstream_addr;
+        ';
+    }
+--- request
+GET /t
+
+--- response_body
+Upstream foo.com
+    Primary Peers
+        127.0.0.1:12354 up
+        127.0.0.1:12355 up
+        127.0.0.1:12357 up
+        127.0.0.1:12358 up
+    Backup Peers
+        127.0.0.1:12356 up
+
+Upstream bar.com
+    Primary Peers
+        127.0.0.1:12354 up
+        127.0.0.1:12355 up
+        127.0.0.1:12357 up
+        127.0.0.1:12359 up
+    Backup Peers
+        127.0.0.1:12356 up
+upstream addr: 127.0.0.1:12354
+upstream addr: 127.0.0.1:12355
+upstream addr: 127.0.0.1:12357
+upstream addr: 127.0.0.1:12358
+
+--- no_error_log
+[error]
+[alert]
+[warn]
+was checked to be not ok
+failed to run healthcheck cycle
+--- error_log
+healthcheck: spawn a thread checking primary peers 0 to 1
+healthcheck: check primary peers 2 to 3
+healthcheck: check primary peer 0
+healthcheck: check backup peer 0
+healthcheck: peer 127.0.0.1:12354 was checked to be ok
+healthcheck: peer 127.0.0.1:12355 was checked to be ok
+healthcheck: peer 127.0.0.1:12357 was checked to be ok
+healthcheck: peer 127.0.0.1:12358 was checked to be ok
+healthcheck: peer 127.0.0.1:12356 was checked to be ok
+healthcheck: peer 127.0.0.1:12359 was checked to be ok
+
+--- timeout: 6
+
+
+=== TEST 27: status_table should return peers with correct properties
```

Comment:
This title is duplicate of the title for `TEST 23`. Maybe prepend "when dedup is true"?